src/tree/assess/TreeAssessor.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Analyzes a paper by doing a list of assessments on a tree representation of a text and its metadata.
+ * Aggregates the scores on each individual assessment into an overall score.
+ *
+ * This score can represent anything from the readability to the SEO of the given text and metadata.
+ *
+ * @memberOf module:tree/assess
+ */
+class TreeAssessor {
+	/**
+	 * Creates a new assessor.
+	 *
+	 * @param {Object} options                                             Assessor options.
+	 * @param {Jed} options.i18n                                           A Jed object to use for translations.
+	 * @param {module:tree/research.TreeResearcher} options.researcher     Supplies the assessments with researches and their (cached) results.
+	 * @param {module:tree/assess.ScoreAggregator} options.scoreAggregator Aggregates the scores on the individual assessments into one overall score.
+	 * @param {module:tree/assess.Assessment[]} [options.assessments]      The list of assessments to apply.
+	 */
+	constructor( options ) {
+		/**
+		 * A Jed object to use for translations.
+		 * @type {Jed}
+		 */
+		this.i18n = options.i18n;
+		/**
+		 * Supplies the assessments with researches and their (cached) results.
+		 * @type {module:tree/research.TreeResearcher}
+		 */
+		this.researcher = options.researcher;
+		/**
+		 * Aggregates the scores on the individual assessments into one overall score.
+		 * @type {module:tree/assess.ScoreAggregator}
+		 */
+		this.scoreAggregator = options.scoreAggregator;
+		/**
+		 * The list of assessments to apply.
+		 * @type {module:tree/assess.Assessment[]|Array}
+		 */
+		this._assessments = options.assessments || [];
+	}
+
+	/**
+	 * Returns the list of available assessments.
+	 *
+	 * @returns {module:tree/assess.Assessment[]} The list of all available assessments.
+	 */
+	getAvailableAssessments() {
```

Comment:
I am not renaming this for now, because of possible backward compatibility reasons, like you mentioned.

Reply:
There are no backward compatibility concerns, because we are introducing a new class. Because of the async methods the places where an `Assessor` is currently used this cannot be replaced by the `TreeAssessor`, so this class is completely new.